test/test_autodiff.cc

```diff
@@ -0,0 +1,46 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include <cassert>
+#include <iostream>
+#include <string>
+#include <unordered_set>
+#include <vector>
+
+#include <gflags/gflags.h>
+#include <glog/logging.h>
+#include <gtest/gtest.h>
+
+#include "tc/core/autodiff.h"
+
+#include "lang_utils.h"
+
+using namespace tc;
+
+void generalTest() {
```

Comment:
I am curious about two things atm so far from looking at the test case:

1. how do we handle temporaries like the example https://facebookresearch.github.io/TensorComprehensions/framework/pytorch_integration/autograd_with_tc.html#reordering-grad-outputs ? would it be possible to demonstrate such use case with test example?

2. How is the inputs/outputs order handled for backwards?

Reply:
#### Regarding 1 and 2:

The test case has temporaries, so you can actually see this in the example. In general the grad TC currently takes in:
- inputs
- outputs (including intermediates)
- grads w.r.t. outputs (including intermediates)
and returns grads w.r.t. inputs. Inputs and outputs are ordered in the same way as they were specified in the original TC.

#### Regarding 3.

No, that's not currently possible, but is a very straightforward extension (only need a few lines of code).